Dockerfile

```diff
@@ -9,8 +9,37 @@ RUN apk add --no-cache npm \
   && npm i /src \
   && npm run build
 
-FROM nginx:latest
+################################################
 
-COPY --from=build  /src/build /usr/share/nginx/html
+FROM alpine:3 as lighttpd-builder
+ARG LIGHTTPD_VERSION="1.4.77"
+# inspired by source build description:https://git.lighttpd.net/lighttpd/lighttpd1.4/src/branch/master/INSTALL
 
-ENTRYPOINT ["/bin/bash", "-c", "/usr/share/nginx/html/generate_config.sh && nginx -g \"daemon off;\""]
+ADD https://download.lighttpd.net/lighttpd/releases-1.4.x/lighttpd-$LIGHTTPD_VERSION.tar.gz \
+  /tmp/lighttpd.tar.gz
+
+RUN apk add \
+  bsd-compat-headers \
+  build-base \
+  pcre2-dev \
+  scons \
+  && tar -C /tmp -xf /tmp/lighttpd.tar.gz \
+  && cd /tmp/lighttpd-$LIGHTTPD_VERSION \
+  && scons -j 4 build_fullstatic=1 build_dynamic=0
```

Comment:
considering build, we might calculate parallel jobs based on available CPUs, like so 🤓 :

```
jobs=$(( $(cat /proc/cpuinfo | grep processor | wc -l) - 2))
```

Reply:
about the mentioned buildcfg you linked:
`zlib                # zlib       ./configure --with-zlib      # (default)`
-> will the build omit libz it is not present (is it instaslled via pcre2 / build-base?)
I think it will make sense to enable compression for lighttpd for performance reasons (hence I am asking)

also, why not use autogen / configure / make instead of scons? 🤔 . in the end, it is a matter of choice I guess. however, considering scons runs on python, this will likely make the installation heavier compared to autogen + make (if we split into separate build, this is of course even less of a concern)

check [archwiki](https://wiki.archlinux.org/title/Lighttpd#Output_compression) for documentation of enabling compression

Reply:
I added both build variants to Makefile. I was not able to get static build via make working with (zlib or deflate) compression yet. We could use scon build for now, it is working as expected.